File: src/main/java/ast/ASTForStmnt.java
package ast;

public class ASTForStmnt extends ASTNode {
    ASTNode inital;
    ASTExpr cond;
    ASTExpr end;
    ASTNodeArray<ASTNode> code;

    public ASTForStmnt(ASTFileLocation loc, ASTNode initial, ASTExpr cond, ASTExpr end, ASTNodeArray<ASTNode> code) {
        super(loc);
        this.inital = initial;
        this.cond = cond;
        this.end = end;
        this.code = code;
    }

    public void print(int i) {
        printIndent(i);
        System.out.println("ASTForStmnt: (initial, cond, end, code)");
        inital.print(i+1);
        cond.print(i+1);
        end.print(i+1);
        code.print(i+1);
    }
}

File: src/main/kotlin/ast/ASTForStmnt.kt
package ast

class ASTForStmnt(loc: ASTFileLocation, val inital: ASTNode, val cond: ASTExpr, val end: ASTExpr, val code: ASTNodeArray<ASTNode>) : ASTNode(loc), ASTBlockStmnt {

    override fun print(i: Int) {
        printIndent(i)
        println("ASTForStmnt: (initial, cond, end, code)")
        inital.print(i + 1)
        cond.print(i + 1)
        end.print(i + 1)
        code.print(i + 1)
    }

    override fun getBlock(): ASTNodeArray<ASTNode> {
        return code
    }
}